### DISCUSSION TITLE
IOS 16.2 - Change of Architecture

### DISCUSSION THREAD

**Discussion by helo-head** [HomeSpan/HomeSpan#417] [Q&A]
I was surprised to hear ( [9to5Mac](https://9to5mac.com/2022/10/25/ios-16-2-new-architecture-home-app/)) that IOS 16.2 will bring a new Homekit architecture. Apparently the user will be prompted on first boot of 16.2 if he/she wants to update their  home to the new architecture. If they choose to upgrade then **_ALL_** clients must be running 16.2. Im hoping this is a beta limitation.

Im assuming this has more to do with managing the home than devices and/or bridges otherwise Apple would have made a bigger deal out this then they have thus far. 

Any idea what this will mean if anything for Homespan?

**Comment by HomeSpan** [comment]
Thanks for noting this.  I'm unsure how this will impact HomeSpan, but based on the article it seems this is more of a back-end update between the Home App and Home Pods.  Unless Apple adds something to HomeKit that disallows non-commercial devices, my guess is HomeSpan will continue to work.

That said, I have a few hundred devices (lots of Lutron dimmers and switches) and have been using HomeKit for about 6 years.  The architecture is in terrible need of an overhaul.  The probability that Siri turns off all the lights when I say "good night" used to be 100% - works flawlessly for about two years.  But with each update to iOS, HomeKit functionality has degraded and now I need to ask Siri a few times to run off all the lights.  Even then sometimes I still find one random light on.  Hopefully this update will make HomeKit a lot more stable and reliable.

**Reply by hblasins** [reply]
Unfortunately it may not be the case. I've just updated the HomeKit architecture, and after the update my HomeSpan device stopped being detected by HomeKit. There seems to be some kind of pairing issue:

** Client #0 Connected: (206 sec) 192.168.88.115 on Socket 3/16
In Pair Setup...** Client #1 Connected: (206 sec) 192.168.88.137 on Socket 4/16

*** ERROR: Device not yet paired!

In Pair Setup...
*** ERROR: Out-of-Sequence Pair-Setup request!

I tried unpairing and pairing the device again. The process seems to complete successfully (device tiles appear in the HomeKit app), but the error message:

*** ERROR: Device not yet paired!

continues to appear.

**Reply by HomeSpan** [reply]
Try unpairing the device from the Home App, then type 'H' into the Serial Monitor to reset the device ID, and then try pairing again.

**Reply by hblasins** [reply]
Unfortunately re-setting the device ID didn't help. I can add all the accessories just fine and the tiles show up in my Home app, but a warning about unresponsive devices is displayed, and I keep seeing this error:
** Client #0 Connected: (210 sec) 192.168.88.240 on Socket 3/16

*** ERROR: Device not yet paired!

** Client #0 Connected: (257 sec) 192.168.88.240 on Socket 4/16

*** ERROR: Device not yet paired!

**Reply by danilkorotkov** [reply]
Some HS devices don’t pair successfully while serial connection is active.

**Comment by HomeSpan** [comment]
I'm going to upgrade my own setup this weekend and will be able to test.  Are you seeing the issue with any other HomeKit devices (Homebridge, commercial devices, etc.).  Also, can you post the full HomeSpan log file?  You can activate it by typing "L2" into the Serial Monitor, or placing `homeSpan.setLogLevel(2)` in your sketch.

**Reply by hblasins** [reply]
Sure thing (see below), I'm happy to run more tests on my side :)
I have a few smart thermostats from Eve (with official Home Kit support) and those work fine. I also use several Shelly 2.5 switches which have a custom firmware with unofficial Home Kit support (https://github.com/mongoose-os-apps/shelly-homekit). Those also behave correctly.

Not sure if this is relevant, but my HomeSpan system only controls Somfy shades/blinds.

`=======================================
** Client #0 Connected: (6410 sec) 192.168.88.240 on Socket 3/16

<<<<<<<<< 192.168.88.240 <<<<<<<<<
POST /pair-verify HTTP/1.1
Host: Somfy-HomeSpan._hap._tcp.local
Content-Length: 37
Content-Type: application/pairing+tlv8
------------ END BODY! ------------
STATE(1) 01
PUBKEY(32) 660B6029C37F457D63A9A5B014F3F25F0D4E6FDF2746AFDC5ACD386CE17EF378
------------ END TLVS! ------------
In Pair Verify #0 (192.168.88.240)...
*** ERROR: Device not yet paired!


>>>>>>>>>> 192.168.88.240 >>>>>>>>>>
HTTP/1.1 200 OK
Content-Type: application/pairing+tlv8
Content-Length: 6

STATE(1) 02
ERROR(1) 01
------------ SENT! --------------

=======================================
** Client #0 Connected: (6584 sec) 192.168.88.240 on Socket 4/16

<<<<<<<<< 192.168.88.240 <<<<<<<<<
POST /pair-verify HTTP/1.1
Host: Somfy-HomeSpan._hap._tcp.local
Content-Length: 37
Content-Type: application/pairing+tlv8
------------ END BODY! ------------
STATE(1) 01
PUBKEY(32) 3DB7F990483400546888D1BA0551CE365C59AEF1598391CF4C70A6C2C4722D10
------------ END TLVS! ------------
In Pair Verify #0 (192.168.88.240)...
*** ERROR: Device not yet paired!


>>>>>>>>>> 192.168.88.240 >>>>>>>>>>
HTTP/1.1 200 OK
Content-Type: application/pairing+tlv8
Content-Length: 6

STATE(1) 02
ERROR(1) 01
------------ SENT! --------------

=======================================
** Client #0 Connected: (6584 sec) 192.168.88.240 on Socket 3/16

<<<<<<<<< 192.168.88.240 <<<<<<<<<
POST /pair-verify HTTP/1.1
Host: Somfy-HomeSpan._hap._tcp.local
Content-Length: 37
Content-Type: application/pairing+tlv8
------------ END BODY! ------------
STATE(1) 01
PUBKEY(32) 15589E70A42803999C6FDCFBAF46D2F72869D3D9D020DB139CCFB0A62A33C827
------------ END TLVS! ------------
In Pair Verify #0 (192.168.88.240)...
*** ERROR: Device not yet paired!


>>>>>>>>>> 192.168.88.240 >>>>>>>>>>
HTTP/1.1 200 OK
Content-Type: application/pairing+tlv8
Content-Length: 6

STATE(1) 02
ERROR(1) 01
------------ SENT! --------------
** Disconnecting Client #0  (6584 sec)
`

**Reply by HomeSpan** [reply]
Sorry, I meant the log files during pairing.  The error message above is indicating the device was not paired.  Since the Home App thinks the device is paired, this is puzzling.  The log file from the pairing process may help debug.

**Reply by HomeSpan** [reply]
Also, can you confirm you are using the latest version, 1.7.0, of HomeSpan.  There was an edge-case issue related to this problem that I had (thought) I fixed.

**Reply by oleksiikutuzov** [reply]
> There was an edge-case issue related to this problem that I had (thought) I fixed.

I can say that I have a bunch of devices running HomeSpan and since 1.7.0 had no issues with the new architecture.

**Reply by hblasins** [reply]
You were right I was using 1.6.0. After upgrading to 1.7.0 the issue is resolved!

**Reply by HomeSpan** [reply]
Thanks for the update.  Really glad it's working now!
